patroni/ha.py

```diff
@@ -25,6 +26,7 @@ def __init__(self, patroni):
         self.old_cluster = None
         self.recovering = False
         self._async_executor = AsyncExecutor()
+        self._disable_sync = False
```

Comment:
I worked on the premise that setting a boolean flag is atomic. That is probably a mix of laziness and premature optimization. However to get the "right value" a mutex on disable sync will not be sufficient or in any way helpful. Instead touch_member needs to be protected by a mutex to avoid getting the value overwritten by stale one.

However I don't think it's worth the effort in this case - it would be better to add a timeout for while_not_sync_standby (2*loop_wait should be enough). Even if there is a race that makes master not notice in time, at worst there will be a small hickup on commits in the master. A timeout is much more robust solution than a locking scheme.

There is a second cluster wide race there anyway:

* Master picks standby to be sync
* Standby publishes nosync
* Standby checks sync state and thinks it can shut down.
* Master makes standby sync standby and publishes sync state
* Standby shuts down
* Master takes up to loop_wait to notice this.

It would be possible to guarantee no unnecessary sync standby waits with a considerable cost in complexity by adding a sync standby candidate field to sync state, have the master publish that and then run an API call to double check before calling synchronous_standby. Thoughts?